src/Native/System.Net.Security.Native/pal_gssapi.h

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+#include "pal_types.h"
+
+#ifdef  GSSFW
+#include <GSS/GSS.h>
+#else
+#include <gssapi/gssapi_ext.h>
+#include <gssapi/gssapi_krb5.h>
+#endif
+
+
+enum GssStatus : uint32_t
+{
+    PAL_GSS_COMPLETE = 0,
+    PAL_GSS_CONTINUE_NEEDED = 1
+};
+
+/*
+Shims the gss_release_buffer method.
+*/
+extern "C" uint32_t GssReleaseBuffer(uint32_t* minor_status, gss_buffer_t buffer);
```

Comment:
gss_buffer_t is a pointer and the managed code uses a SafeHandle to pass this parameter to the shim. However there is one assumption about gss_buffer_desc structure in managed code which @kapilash is working on removing.